BraveWallet/Crypto/Stores/SwapTokenStore.swift

```diff
@@ -32,48 +33,384 @@ public class SwapTokenStore: ObservableObject {
+      self?.updatingPriceQuote = false
+    }
   }
   
-  func prepare(with accountInfo: BraveWallet.AccountInfo) {
+  private func generateSwapParams(_ base: SwapParamsBase) -> BraveWallet.SwapParams? {
+    guard
+      let accountInfo = accountInfo,
+      let sellToken = selectedFromToken,
+      let buyToken = selectedToToken
+    else { return nil }
+    
+    let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: 18))
+    let sellAddress = sellToken.isETH ? BraveWallet.ethSwapAddress : sellToken.contractAddress
+    let buyAddress = buyToken.isETH ? BraveWallet.ethSwapAddress : buyToken.contractAddress
+    let sellAmountInWei: String
+    let buyAmountInWei: String
+    switch base {
+    case .perSellAsset:
+      sellAmountInWei = weiFormatter.weiString(from: sellAmount, radix: .decimal, decimals: Int(sellToken.decimals)) ?? "0"
+      buyAmountInWei = ""
+    case .perBuyAsset:
+      sellAmountInWei = ""
+      buyAmountInWei = weiFormatter.weiString(from: buyAmount, radix: .decimal, decimals: Int(buyToken.decimals)) ?? "0"
+    }
+    let swapParams = BraveWallet.SwapParams(
+      takerAddress: accountInfo.address,
+      sellAmount: sellAmountInWei,
+      buyAmount: buyAmountInWei,
+      buyToken: buyAddress,
+      sellToken: sellAddress,
+      slippagePercentage: slippage,
+      gasPrice: ""
+    )
+    
+    return swapParams
+  }
+  
+  private func clearAllAmount() {
+    sellAmount = "0"
+    buyAmount = "0"
+    selectedFromTokenPrice = "0"
+  }
+  
+  private func updateSwapControls(swapResponse: BraveWallet.SwapResponse?, base: SwapParamsBase) {
+    guard let respnose = swapResponse else { return }
+    let weiFormatter = WeiFormatter(decimalFormatStyle: .decimals(precision: 18))
+    switch base {
+    case .perSellAsset:
+      var decimal = 18
+      if let fromToken = selectedFromToken {
+        decimal = Int(fromToken.decimals)
+      }
+      let decimalString = weiFormatter.decimalString(for: respnose.buyAmount, decimals: decimal) ?? ""
+      if let bv = BDouble(decimalString) {
+        buyAmount = bv.decimalDescription
+      }
+    case .perBuyAsset:
+      var decimal = 18
+      if let toToken = selectedToToken {
+        decimal = Int(toToken.decimals)
+      }
+      let decimalString = weiFormatter.decimalString(for: respnose.sellAmount, decimals: decimal) ?? ""
+      if let bv = BDouble(decimalString) {
+        sellAmount = bv.decimalDescription
+      }
+    }
+    
+    if let bv = BDouble(respnose.price) {
+      // will need to invert price if price quote is based on buyAmount
+      // ref from slack:
+      // https://bravesoftware.slack.com/archives/C023VS4HJ6Q/p1636579425364500?thread_ts=1636570735.354500&cid=C023VS4HJ6Q
+      let price = base == .perSellAsset ? bv : 1 / bv
+      selectedFromTokenPrice = price.decimalDescription
+    }
+    
+    checkBalanceShowError(swapResponse: respnose)
+  }
+  
+  private func activeErc20Allowance(_ spenderAddress: String) {
```

Comment:
Nit: Naming is confusing, needs new name or doc, this function _creates_ the transaction after making approval data to send in right? The enum is named `activateAllowance`, so is this supposed to be `activate` as well? Maybe something simpler like `createERC20SwapTransaction` would be more clear